### PR TITLE
flounder core fixes/changes

### PR DESCRIPTION
**Pull request by damian-rodriguez-sociomantic**
this PR includes fixes for [this](https://github.com/sociomantic-tsunami/flounder-react/issues/8) and [this](https://github.com/sociomantic-tsunami/flounder-react/issues/7).

Also, we've made some changes to the search input to make it display correctly when we have multiple tags selected.

Tests are also up to date

### CODE REVIEW

**Comment by mousemke** [comment]
so, this looks amazing.  but.....  it's forked from master.  that's makes life very difficult.  1.0.0 is being worked on heavily (see the first line of the readme), so this needs to be merged into `flounder1.0.0`  which is much more of a challenge coming out of `master` 
exhibit A

the fact that there has been big changes to multiTags will make this even more difficult.

i dont really know what to suggest.  i guess either you need to start in `flounder1.0.0`, or I need to find time to merge this.  This PR will not be merged into `0.8.5`, it's frozen as it doesnt make sense to update problems in 2 branches.

I do appreciate the PRs, but this is the second time it's been a master PR.  Is there anything i can do to make things more clear?

**Comment by josh-wilson-sociomantic** [comment]
@mousemke seems like we need to stick that info in the readme if master is not always at latest release state.

This is confusing though, as master notes it's at version 0.8.5 in the readme. Which is not the case it seems...

I'd opt for us keeping master clean and then it's simple to fork / make PRs, as opposed to having to fork a specific version/branch before making changes.

And then any developments could be done on specific feature branches etc.

For me, that would seem like a good setup for accepting PRs and keeping merge conflicts to a minimum.

@zsolt-nagy-sociomantic , you have any thoughts on this?

----

Regarding this set of changes, do you have an ETA for v1? Probably makes sense for us to make changes off of this branch to keep merge conflicts to a minimum. 

So where exactly would be the desired place to branch from before applying changes? master at it's current state? Or you're saying to wait for 1 to drop and then fire back in with the PRs?

**Comment by mousemke** [comment]
1.0.0 has [changed a lot](https://github.com/sociomantic-tsunami/flounder/compare/master...flounder1.0.0?expand=1).  anything on master is locked and has a git and an npm release. dev will never be master.  master is in fact the current release state; that's the exact reason that a pr into it isnt good.  

from the master readme first line:
`*** Flounder 1.0.0 is being built right now, so there is a feature freeze on earlier versions, but feel free to open issues for 0.X.X and feature issues for 1.0.0 should anything come up ***`.  There is also a 1.0.0 milestone open.  this is only a bigger issue this time around due to the incompatibility in the branches (hence the semver)

you could fork `flounder1.0.0` and change it there, or wait on 1.0.0 to drop, your call.

ETA is hard. i want to get 100% test coverage before releasing, but if y'all need it, that can be skipped possibly.  Additionally, the test suite is moving from qunit to mocha and istanbul.  so it's a fairly large issue.

TL;DR - I could get you a stable 1.0.0 branch, but i probably wouldnt release 1.0.0 until tests are ready

**Comment by josh-wilson-sociomantic** [comment]
@mousemke Cool. That's fine, thanks for the infos 👍 

Let's add your desired PR workflow to the readme though to avoid future confusion?

**Comment by mousemke** [comment]
@josh-wilson-sociomantic definately will do

**Comment by mousemke** [comment]
ok.  for now, can we say PR into `flounder1.0.0`

after, i will add details to the PR process in the readme.  thinking something like:

branches
-----

+ release - exactly the same as npm and git releases (including dist files)
+ master - stable, but no dist files
+ dev - dev that all should branch out from
+ feature branches - specific features/fixes, branched from dev and merging back in when complete


i'm open to other ideas on organization as well.

**Comment by mousemke** [comment]
untitled_2016-07-08_17-50-56

@josh-wilson-sociomantic @damian-rodriguez-sociomantic @zsolt-nagy-sociomantic
